weblate/screenshots/models.py

```diff
@@ -76,3 +91,53 @@ def change_screenshot_assignment(sender, instance, action, **kwargs):
         name="UnusedScreenshot"
     ).exists():
         component_alerts.delay([instance.pk])
+
+
+def validate_screenshot_image(component, filename):
+    """Returns True if image is validated."""
+    try:
+        full_name = os.path.join(component.full_path, filename)
+        with open(full_name, "rb") as f:
+            image_file = File(f, name=os.path.basename(filename))
+            validate_bitmap(image_file)
+    except ValidationError:
+        report_error(cause="Failed to validate image from repository")
+        return False
+    return True
+
+
+@receiver(vcs_post_update)
+def update_screenshots_from_repo(sender, component, previous_head: str, **kwargs):
+    repository = component.repository
+    changed_files = repository.get_changed_files(compare_to=previous_head)
+
+    screenshots = Screenshot.objects.filter(
+        translation__component=component, repository_filename__in=changed_files
+    )
+
+    # Update existing screenshots
+    for screenshot in screenshots:
```

Comment:
Can you please add tests for screenshot update from repo as well? This seems not tested at all.